src/common/file_buffer.cpp

```diff
@@ -1,45 +1,75 @@
 #include "duckdb/common/file_buffer.hpp"
-#include "duckdb/common/file_system.hpp"
-#include "duckdb/common/helper.hpp"
+
+#include "duckdb/common/allocator.hpp"
 #include "duckdb/common/checksum.hpp"
 #include "duckdb/common/exception.hpp"
-#include "duckdb/common/allocator.hpp"
+#include "duckdb/common/file_system.hpp"
+#include "duckdb/common/helper.hpp"
 
 #include <cstring>
 
 namespace duckdb {
 
 FileBuffer::FileBuffer(Allocator &allocator, FileBufferType type, uint64_t bufsiz)
     : allocator(allocator), type(type), malloced_buffer(nullptr) {
-	const int sector_size = Storage::SECTOR_SIZE;
-	// round up to the nearest sector_size, this is only really necessary if the file buffer will be used for Direct IO
-	if (bufsiz % sector_size != 0) {
-		bufsiz += sector_size - (bufsiz % sector_size);
-	}
-	D_ASSERT(bufsiz % sector_size == 0);
-	D_ASSERT(bufsiz >= sector_size);
-	// we add (sector_size - 1) to ensure that we can align the buffer to sector_size
-	malloced_size = bufsiz + (sector_size - 1);
+	SetMallocedSize(bufsiz);
 	malloced_buffer = allocator.AllocateData(malloced_size);
+	Construct(bufsiz);
+}
+
+void FileBuffer::SetMallocedSize(uint64_t &bufsiz) {
+	// make room for the block header (if this is not the db file header)
+	if (type == FileBufferType::MANAGED_BUFFER && bufsiz != Storage::FILE_HEADER_SIZE) {
+		bufsiz += Storage::BLOCK_HEADER_SIZE;
```

Comment:
The storage header size is normally only required for blocks read from disk - as it contains the checksum of those blocks. Should this be added to every managed buffer?

Reply:
We resolved this off-line, but for anyone reading this: managed buffers can be offloaded to disk, and their checksum will also be checked, so therefore every managed buffer needs this.